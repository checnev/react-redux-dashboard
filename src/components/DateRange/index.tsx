import React from 'react';
import { TextField, TextFieldProps } from '@material-ui/core';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) => createStyles({
  root: {},
  input: {
    width: 150,
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
  },
}));

interface DateRangeProps {
  from: TextFieldProps;
  to: TextFieldProps;
}
const DateRange: React.FC<DateRangeProps> = ({ from, to }: DateRangeProps) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <TextField
        label={from.label || 'From'}
        type="date"
        name="from"
        defaultValue={from.defaultValue}
        className={classes.input}
        onChange={from.onChange}
      />
      <TextField
        label={to.label || 'To'}
        type="date"
        name="to"
        defaultValue={to.defaultValue}
        className={classes.input}
        onChange={to.onChange}
      />
    </div>
  );
};

export default DateRange;
