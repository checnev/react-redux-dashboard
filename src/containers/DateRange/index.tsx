import { useDispatch, useSelector, shallowEqual } from 'react-redux';
import { setFromDate, setToDate } from 'actions/dashboard';
import { dateRangeSelector } from 'selectors/dashboard';
import DateRange from 'components/DateRange';


const DateRangeContainer = () => {
  const dispatch = useDispatch();
  const dateRange = useSelector(dateRangeSelector, shallowEqual);

  const changeDate = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (event.target.name === 'from') {
      dispatch(setFromDate(event.target.value));
    } else {
      dispatch(setToDate(event.target.value));
    }
  };

  return <DateRange 
    from={{
      defaultValue: dateRange.from.format('YYYY-MM-DD'),
      onChange: changeDate,
    }}
    to={{
      defaultValue: dateRange.to.format('YYYY-MM-DD'),
      onChange: changeDate,
    }}
  />;
};

export default DateRangeContainer;
