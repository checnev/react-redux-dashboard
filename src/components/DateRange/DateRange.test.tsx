import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from 'pretty';
import DateRange from 'components/DateRange';

let container: any = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('DateRange render', () => {
  const props = {
    from: {
      label: 'From label',
      defaultValue: '01.01.1971',
      onChange: () => { },
    },
    to: {
      label: 'To label',
      defaultValue: '01.01.1971',
      onChange: () => { },
    },
  };

  act(() => {
    render(<DateRange {...props} />, container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot(); 
});
