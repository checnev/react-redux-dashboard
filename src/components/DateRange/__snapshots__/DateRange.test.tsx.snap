// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRange render 1`] = `
"<div class=\\"makeStyles-root-1\\">
  <div class=\\"MuiFormControl-root MuiTextField-root makeStyles-input-2\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\">From label</label>
    <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" name=\\"from\\" type=\\"date\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"01.01.1971\\"></div>
  </div>
  <div class=\\"MuiFormControl-root MuiTextField-root makeStyles-input-2\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\">To label</label>
    <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" name=\\"to\\" type=\\"date\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"01.01.1971\\"></div>
  </div>
</div>"
`;
